src/server/pfs/server/driver.go

```diff
@@ -658,6 +627,35 @@ func (d *driver) makeCommit(ctx context.Context, parent *pfs.Commit, branch stri
 			}
 			commitInfo.ParentCommit = parent
 		}
+
+		// Use a map to de-dup provenance
```

Comment:
How come this block got moved farther down?

Reply:
It has to be below the code that figures out / fills in the ParentCommit. The `appendSubvenance` call below takes `commitInfo` as an argument and it needs that `commitInfo` to have the `ParentCommit` field filled in so that it can figure out of one of the CommitRanges in the subvenance has its parent as the upper bound.